storage: remove {Has,Put}BogusSideloadedData

tbg opened the pull request:
Those were a pain to deal with while looking into Raft log size
computations. They're also ugly, so get rid of them.

One call was simply removed - we can do without this coverage now that
the replicaID does not influence the sideloaded storage location any
more (#35035).

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/35094)
<!-- Reviewable:end -->

petermattis reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/35094#-:-LZAPPzq5k7EAAIcw9Ul:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @petermattis and @tbg)

___
*[pkg/storage/client_replica_gc_test.go, line 93 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/35094#-LZAPFt2-IUDEzan3z7B:-LZAPFt3E5c4NtvdLUen:b-wxhmzl) ([raw file](https://github.com/cockroachdb/cockroach/blob/5d73ed35275c9168c37963d0919c86b9e5e886ac/pkg/storage/client_replica_gc_test.go#L93)):*
> ```Go
> 		); err != nil {
> 			t.Fatal(err)
> 		}
> ```

I think you should check that `repl1.SideloadedDir()` is not empty at this point, otherwise the test could trivially pass.


<!-- Sent from Reviewable.io -->

tbg committed:
storage: remove {Has,Put}BogusSideloadedData

Those were a pain to deal with while looking into Raft log size
computations. They're also ugly, so get rid of them.

One call was simply removed - we can do without this coverage now that
the replicaID does not influence the sideloaded storage location any
more (#35035).

Release note: None
